src/common/strtol.cc

```diff
@@ -16,18 +16,30 @@
 
 #include <climits>
 #include <limits>
+#include <cmath>
 #include <sstream>
+#include <string_view>
 
 using std::ostringstream;
 
-long long strict_strtoll(const char *str, int base, std::string *err)
+long long strict_strtoll(const std::string_view str, int base, std::string *err)
 {
-  char *endptr;
   std::string errStr;
+  auto invalid = str.find_first_not_of("0123456789-+");
+  if (invalid != std::string_view::npos ||
+      invalid == 0) {
+    errStr = "The option value '";
+    errStr.append(str);
+    errStr.append("'");
+    errStr.append(" contains invalid digits");
+    *err =  errStr;
+    return 0;
+  }
+  char *endptr;
   errno = 0; /* To distinguish success/failure after call (see man page) */
-  long long ret = strtoll(str, &endptr, base);
+  long long ret = strtoll(str.data(), &endptr, base);
```

Comment:
Looks good, but note that string_view::data() may not be NULL-terminated. I don't see that happening here, but if you want to be /extra/ pedantic, checking might be appropriate.